tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -2723,22 +2723,32 @@ namespace hlsl {
 
     bool VisitDeclRefExpr(DeclRefExpr *ref) {
       ValueDecl *valueDecl = ref->getDecl();
-      FunctionDecl *fnDecl = dyn_cast_or_null<FunctionDecl>(valueDecl);
-      fnDecl = getFunctionWithBody(fnDecl);
-      if (fnDecl) {
+      RecordFunctionDecl(dyn_cast_or_null<FunctionDecl>(valueDecl));
+      return true;
+    }
+
+    bool VisitCXXMemberCallExpr(CXXMemberCallExpr* callExpr)
+    {
+      RecordFunctionDecl(callExpr->getMethodDecl());
+      return true;
+    }
+
+    void RecordFunctionDecl(FunctionDecl* funcDecl)
+    {
+      funcDecl = getFunctionWithBody(funcDecl);
+      if (funcDecl) {
         if (m_sourceIt == m_callNodes.end()) {
           auto result = m_callNodes.insert(
-            std::pair<FunctionDecl *, CallNode>(m_source, CallNode{ m_source }));
+            std::pair<FunctionDecl*, CallNode>(m_source, CallNode{ m_source }));
```

Comment:
you don't use make_pair for this sort of stuff ?

Reply:
Yeah, I would have. This line was only changed because of the autoformatter.